### PR TITLE
Fix enum map and extension

### PR DESCRIPTION
**Pull request by luca-vz**
Generating enums with an itemType int/double created maps and extensions where the itemType was still a String, this has been changed. Besides that the name of xMapping of an enum was first generated as XMapping and is now getting generated correctly (xMapping).

### CODE REVIEW

**Comment by ikbendewilliam** [comment]
Next time add an issue with the details of what went wrong

**Comment by ikbendewilliam** [comment]
LGTM
Sidenote, create a new issue for the deprecated lint rule and `ignore` field
```
   info • 'invariant_booleans' is a deprecated lint rule and should not be used
          • example/analysis_options.yaml:58:7 • deprecated_lint_hint
   info • 'ignore' is deprecated and shouldn't be used. Use `includeFromJson`
          and `includeToJson` with a value of `false` instead •
          example/lib/model/user/testing.dart:13:55 • deprecated_member_use
```

File: lib/util/case_util.dart [hunk 1]
```diff
@@ -41,21 +41,25 @@ class CaseUtil {
     return words;
   }
 
-  String get snakeCase => _getSnakeCase();
+  String get lowerCamelCase => _getLowerCamelCase();
 
-  String get camelCase => _getCamelCase();
+  String get upperCamelCase => _getUpperCamelCase();
```
Comment (vanlooverenkoen) [inline lib/util/case_util.dart:46]: There is no such thing as upperCamelCase. This is just PascalCase

**Review by vanlooverenkoen — CHANGES_REQUESTED** [review]

**Comment by vanlooverenkoen** [comment]
Ticket created for the ignore: https://github.com/icapps/flutter-model-generator/issues/134

**Pull request merged**